lib/charms/observability_libs/v0/kubernetes_compute_resources_patch.py

```diff
@@ -569,6 +569,7 @@ def __init__(
         *,
         resource_reqs_func: Callable[[], ResourceRequirements],
         refresh_event: Optional[Union[BoundEvent, List[BoundEvent]]] = None,
+        allow_patching_charm_container: bool = False
```

Comment:
Given the position of the juju team, the lib should not act on "charm" and rather fail.

Reply:
but what if the user has an `assumes: juju<3.6.8`?